### PR TITLE
Just validation stuff

### PR DESCRIPTION
**Pull request by dakota002**

### CODE REVIEW

**Comment by dakota002** [comment]
The $id fields need to be sorted out, I will make a separate PR if Vidushi hasn't gotten to it already

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: validate.mjs [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
+  const schemas = await Promise.all(
+    filesInDir.map(async (file) =>
+      JSON.parse(
+        await readFile(file, {
+          encoding: 'utf-8',
+        })
+      )
+    )
+  )
+  ajv.addSchema(schemas).compile(true)
```
Comment (lpsinger) [inline validate.mjs:24]: What happens on error?

Can we please retain the original behavior that:

* if there are no errors, the process exits cleanly
* if there are any validation errors, they are all printed with the format `${path}: ${e.message}`, and then the process exits with status 1
* if there are any errors that are not validation errors, then they are re-thrown

File: validate.mjs [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
+  return true
```
Comment (lpsinger) [inline validate.mjs:25]: No need for a return value:

```suggestion
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: validate.mjs [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
-    return true
+    const schemas = await Promise.all(
+      filesInDir.map(async (file) =>
+        JSON.parse(
+          await readFile(file, {
+            encoding: 'utf-8',
+          })
+        )
+      )
+    )
```
Comment (lpsinger) [inline validate.mjs:24]: Move out of the `try` block.

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Review by Vidushi-GitHub — APPROVED** [review]

**Review by lpsinger — APPROVED** [review]

**Pull request merged**